index.js

```diff
@@ -230,17 +229,37 @@ async function terraform(terraformDirectory, args) {
     process.exit(exitCode);
   }
 
+  let changes = false;
+
   core.startGroup('Run terraform plan');
-  const tfp = await terraform(terraformDirectory, ['plan', `-lock=${terraformLock}`, `-parallelism=${terraformParallelism}`, `-refresh=${terraformRefresh}`, '-out=terraform.plan'].concat(terraformTargets).concat(terraformVariableFiles).concat(terraformPlanDestroy));
+  const tfp = await terraform(terraformDirectory, ['plan', `-lock=${terraformLock}`, `-parallelism=${terraformParallelism}`, `-refresh=${terraformRefresh}`, '-out=terraform.plan', '-detailed-exitcode'].concat(terraformTargets).concat(terraformVariableFiles).concat(terraformPlanDestroy));
   core.endGroup();
-  if (tfp.status > 0) {
+  if (tfp.status == 1) { // see -detailed-exitcode documentation
     tf_plan = status_failed;
     core.setFailed(`Failed to prepare the terraform plan [err:${tfp.status}]`);
     terraformDoApply = false;
   } else {
+    changes = tfp.status == 2;
+    core.setOutput('changes', changes ? 'true' : 'false');
```

Comment:
Can be moved outside of the `else` block? Would also remove the need for initializing it with `false`.